### DISCUSSION TITLE
Auth with external REST api session cookie

### DISCUSSION THREAD

**Discussion by galaczi** [sveltejs/kit#8454] [Q&A]
I am completely new to Svelte and Sveltekit. So far I am in love with it! But I am confused about how authentication and authorization would work with an existing external REST api. The rest api handles authentication and sets a session cookie. The API and the Svelte app are on the same domain.

As far as I know, ``fetch`` forwards cookies inside load on the server, so API requests should work. So is it a matter of simply checking the REST api response and if it's unauthorized, I also handle it in the Svelte route? Is that the big picture?

**Comment by galaczi** [comment]
Anyone knows how it's done?

**Comment by Devr-pro** [comment]
You can pass options in `fetch` which headers are your session cookies.

**Reply by galaczi** [reply]
Thanks for the reply! And based on the status code of the REST api response would I send the response from Svelte Kit I guess? I mean nothing would be persisted in Svelte about the user's session, right?
